tools/python3/git.py

```
@@ -0,0 +1,35 @@
+
+def branch_exists(branch):
+    try:
+        run(f"git show-branch {branch}", check=True)
+    except subprocess.CalledProcessError as e:
+        return False  # exit failure means branch does not exist
+    return True
+
+
+def checkout(branch, create=False, recreate=False):
+    if recreate:
+        delete_branch(branch)
```

Ruff

Undefined name `delete_branch`

(F821)